packages/@expo/fingerprint/src/ProjectWorkflow.ts

```diff
@@ -0,0 +1,224 @@
+  getRootPathAsync(): Promise<string>;
+  isFileIgnoredAsync(filePath: string): Promise<boolean>;
+}
+
+async function getVCSClientAsync(projectRoot: string): Promise<VCSClient> {
+  if (await isGitInstalledAndConfiguredAsync()) {
+    return new GitClient();
+  } else {
+    return new NoVCSClient(projectRoot);
+  }
+}
+
+async function isFileIgnoredByEasIgnoreAsync(
+  projectRoot: string,
+  filePath: string
+): Promise<boolean | undefined> {
+  const easIgnorePath = path.join(projectRoot, EASIGNORE_FILENAME);
+  if (await pathExistsAsync(easIgnorePath)) {
+    const ignore = new Ignore(projectRoot);
+    await ignore.initIgnoreWithEasIgnoreAsync(await fs.readFile(easIgnorePath, 'utf-8'));
+    return ignore.ignores(filePath);
+  }
+  return undefined;
+}
+
+class GitClient implements VCSClient {
+  public async getRootPathAsync(): Promise<string> {
+    return (await spawnAsync('git', ['rev-parse', '--show-toplevel'])).stdout.trim();
+  }
+
+  async isFileIgnoredAsync(filePath: string): Promise<boolean> {
+    const ignoredByEasIgnore = await isFileIgnoredByEasIgnoreAsync(
+      await this.getRootPathAsync(),
+      filePath
+    );
+    if (ignoredByEasIgnore !== undefined) {
+      return ignoredByEasIgnore;
+    }
+
+    try {
+      await spawnAsync('git', ['check-ignore', '-q', filePath], {
+        cwd: path.normalize(await this.getRootPathAsync()),
+      });
+      return true;
+    } catch {
+      return false;
+    }
+  }
+}
+
+class NoVCSClient implements VCSClient {
+  constructor(private readonly projectRoot: string) {}
+
+  async getRootPathAsync(): Promise<string> {
+    return this.projectRoot;
+  }
+
+  async isFileIgnoredAsync(filePath: string): Promise<boolean> {
+    const ignoredByEasIgnore = await isFileIgnoredByEasIgnoreAsync(
+      await this.getRootPathAsync(),
+      filePath
+    );
+    if (ignoredByEasIgnore !== undefined) {
+      return ignoredByEasIgnore;
+    }
+
+    const ignore = new Ignore(this.projectRoot);
+    await ignore.initIgnoreAsync();
+    return ignore.ignores(filePath);
+  }
+}
+
+async function isGitInstalledAndConfiguredAsync(): Promise<boolean> {
+  try {
+    await spawnAsync('git', ['--help']);
+  } catch (error: any) {
+    if (error.code === 'ENOENT') {
+      return false;
+    }
+    throw error;
+  }
+
+  try {
+    await spawnAsync('git', ['rev-parse', '--show-toplevel']);
+  } catch {
+    return false;
+  }
+
+  return true;
+}
+
+const EASIGNORE_FILENAME = '.easignore';
+const GITIGNORE_FILENAME = '.gitignore';
+const DEFAULT_IGNORE = `
+.git
+node_modules
+`;
+
+/**
+ * Ignore wraps the 'ignore' package to support multiple .gitignore files
+ * in subdirectories.
+ *
+ * Inconsistencies with git behavior:
+ * - if parent .gitignore has ignore rule and child has exception to that rule,
+ *   file will still be ignored,
+ * - node_modules is always ignored
+ *
+ * Differs from the eas-cli Ignore class by not using `.easignore`. Otherwise this is copied. May try
```

Comment:
I think this comment indicates that we shouldn't use easignore (which IMO is correct). But below it does use it?